src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts

```diff
@@ -728,6 +728,11 @@ export class ReadingsGraphComponent implements OnDestroy {
         }
       }
     };
+    if(this.optedTime > 86400){
```

Comment:
I would suggest to make it dynamic that if dataset (ticks) length is above a limit then show days only for a week else show day hours